### ISSUE TITLE
Guess type based on FieldDescriptionInterface

### ISSUE THREAD

**Issue by franmomu** [sonata-project/SonataAdminBundle#6701]
#### TL;DR

It would be nice to change from 

`TypeGuesserInterface::guessType(string $class, string $property, ModelManagerInterface $modelManager)` 

to 

`TypeGuesserInterface::guessType(FieldDescriptionInterface $fieldDescription)`

That would allow us to remove some code in the persistence bundles.

#### Long version

Right now `TypeGuesserInterface` is like this:

https://github.com/sonata-project/SonataAdminBundle/blob/ecab71416e9441e1ae839ee1525e3a95049f1763/src/Guesser/TypeGuesserInterface.php#L22-L33

The `guessType` method receives a `$class`, `$property` and a `$modelManager`. This interface is used by the persistence bundles in order to guess which type (for show, list and filters) should be used if not specified based on these properties.

Looking at the implementations of this interface, we have:

In `sonata-project/doctrine-orm-admin-bundle` (similar to `sonata-project/doctrine-mongodb-admin-bundle`):
- [TypeGuesser](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/0c5019b4d7da56476c44e78dd4f2ffbaba06e5dc/src/Guesser/TypeGuesser.php)
- [FilterTypeGuesser](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/0c5019b4d7da56476c44e78dd4f2ffbaba06e5dc/src/Guesser/FilterTypeGuesser.php)
- Both extend from [AbstractTypeGuesser](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/0c5019b4d7da56476c44e78dd4f2ffbaba06e5dc/src/Guesser/AbstractTypeGuesser.php)

In all of this implementations what happens is that it calls the corresponding `modelManager` to retrieve metadata for that class and property (it calls to a `ModelManagerInterface::getParentMetadataForProperty()` method which does not exist in the interface, this method is also used to construct the `FieldDescription`).

Looking at where the interface is used (the same thing for the `mongodb` bundle):
- [DatagridBuilder](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/0c5019b4d7da56476c44e78dd4f2ffbaba06e5dc/src/Builder/DatagridBuilder.php#L138)
- [ListBuilder](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/0c5019b4d7da56476c44e78dd4f2ffbaba06e5dc/src/Builder/ListBuilder.php#L56)
- [ShowBuilder](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/0c5019b4d7da56476c44e78dd4f2ffbaba06e5dc/src/Builder/ShowBuilder.php#L56)

It always has a `FieldDescriptionInterface` instance when calling `guessType` and given that since `sonata-project/admin-bundle` 4.0 an instance of `FieldDescriptionInterface` would have all the mapping information since the construction, I think we could change the `guessType` method to just guess the type based on the `FieldDescriptionInterface` and we can get rid of the non-existing `ModelManagerInterface::getParentMetadataForProperty()` method.

This issue is just in case someone has an opinion or wants to give it a try or something.
